### PR TITLE
Translation 82444 show banner to ask users to update their browsers

### PR DESCRIPTION
**Pull request by iman-fs**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/layout/footer/social-wrapper.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import Instagram from 'images/svg/layout/footer-instagram.svg'
 import Facebook from 'images/svg/layout/footer-facebook.svg'
 import Linkedin from 'images/svg/layout/footer-linkedin.svg'
 
+// test
```
[the remark this answers is not in the document]
Comment (iman-fs) [reply src/components/layout/footer/social-wrapper.tsx:15]: will remove it

File: src/common/constants.ts [hunk 2]
```diff
@@ -152,3 +152,14 @@ export const youtube_url = 'https://www.youtube.com/@deriv'
 export const cms_end_point = 'https://deriv-academy.directus.app'
 export const cms_assets_end_point = `${cms_end_point}/assets/`
 export const loss_percent = 70
+export const browsers_minimum_required_version = {
```
[the remark this answers is not in the document]
Comment (iman-fs) [reply src/common/constants.ts:155]: I agree with you. first I put it inside the hook but @habib-deriv told me to move it to constants. what do you think @habib-deriv ?

File: src/components/layout/modal/browser_update_alert_modal.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const cookie_key = 'browser_update_alert_modal_shown'
+
+const BrowserUpdateAlertModal = () => {
+    const modalRef = React.useRef<ModalRefType>()
+    const { is_outdated } = useUserBrowser()
+    const cookie = new CookieStorage(cookie_key)
+
+    useEffect(
+        () => is_outdated && !cookie.get(cookie_key) && modalRef.current.show(),
+        [is_outdated],
```
[the remark this answers is not in the document]
Comment (iman-fs) [reply src/components/layout/modal/browser_update_alert_modal.tsx:17]: I will remove it

File: src/components/hooks/use-user-browser.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react'
+import { browserName, browserVersion } from 'react-device-detect'
+import { browsers_minimum_required_version as default_versions } from 'common/constants'
+
+export const useUserBrowser = (
+    browsers_minimum_required_version: typeof default_versions = default_versions,
```
[the remark this answers is not in the document]
Comment (iman-fs) [reply src/components/hooks/use-user-browser.tsx:6]: sure. will change the code.

File: src/components/layout/modal/browser_update_alert_modal.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    )
+
+    const handlePositive = () => {
+        const expiration_date = Date.now() + 14 * 86400 * 1000
```
[the remark this answers is not in the document]
Comment (iman-fs) [reply src/components/layout/modal/browser_update_alert_modal.tsx:21]: done

**Review by mitra-deriv — DISMISSED** [review]

File: src/components/elements/modal.tsx [hunk 6]
```diff
@@ -43,7 +43,8 @@ const HeaderContainer = styled(Flex)`
 `
 
 const ContentContainer = styled.div`
-    overflow: scroll;
+    overflow: auto;
+    max-height: 70vh;
```
Comment (habib-deriv) [inline src/components/elements/modal.tsx:47]: Do we need this?

File: src/components/layout/modal/browser_update_alert_modal.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import React, { useEffect } from 'react'
+import { ModalRefType } from 'components/elements/modal'
+import { Localize } from 'components/localization'
+import { Modal, Header } from 'components/elements'
+import { useUserBrowser } from 'components/hooks/use-user-browser'
+import { CookieStorage } from 'common/storage'
+
+const cookie_key = 'browser_update_alert_modal_shown'
```
Comment (habib-deriv) [inline src/components/layout/modal/browser_update_alert_modal.tsx:8]: Please move this variables on constants.ts

File: src/components/layout/modal/browser_update_alert_modal.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+const browsers_minimum_required_version = {
```
Comment (habib-deriv) [inline src/components/layout/modal/browser_update_alert_modal.tsx:9]: Please move this variables on constants.ts

File: src/components/elements/modal.tsx (hunk 6 above)
Comment (iman-fs) [reply src/components/elements/modal.tsx:47]: yes. with `overflow: scroll;` there will be a disabled scrollbar (both on x and y) in linux devices (and maybe windows) even if the content is not big . So I added this to show the scrollbar only if the content is too much that it needs more than 70vh height.

File: src/components/layout/modal/browser_update_alert_modal.tsx (hunk 7 above)
Comment (iman-fs) [reply src/components/layout/modal/browser_update_alert_modal.tsx:8]: done

File: src/components/layout/modal/browser_update_alert_modal.tsx (hunk 8 above)
Comment (iman-fs) [reply src/components/layout/modal/browser_update_alert_modal.tsx:9]: done

**Review by habib-deriv — DISMISSED** [review]
LGTM

File: src/components/layout/modal/browser_update_alert_modal.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import { ModalRefType } from 'components/elements/modal'
+import { Localize } from 'components/localization'
+import { Modal, Header } from 'components/elements'
+import { useUserBrowser } from 'components/hooks/use-user-browser'
+import { CookieStorage } from 'common/storage'
+import { cookie_key, browsers_minimum_required_version } from 'common/constants'
+
+const BrowserUpdateAlertModal = () => {
+    const modalRef = React.useRef<ModalRefType>()
+    const { is_outdated } = useUserBrowser(browsers_minimum_required_version)
```
Comment (prince-deriv) [inline src/components/layout/modal/browser_update_alert_modal.tsx:11]: do we have a situation where we need to pass another config for this? If not can we keep `browsers_minimum_required_version` inside the `useUserBrowser` so we don't need to pass ot like this?

Comment (iman-fs) [reply src/components/layout/modal/browser_update_alert_modal.tsx:11]: Right now we don't have such situation.
At first I kept `browsers_minimum_required_version` inside the `useUserBrowser` but then we decided to change it in a way that everyone be able to pass any config that they want. I can make the param optional and use `browsers_minimum_required_version` inside the hook as the default value for the param. 
what do you think @prince-deriv ?
@ali-hosseini-fs can you give your opinion on this?

**Review by eduard-deriv — APPROVED** [review]

**Comment by eduard-deriv** [comment]
Well Done 👍

**Review by mohammad-rework — APPROVED** [review]

**Pull request merged**